test/test_torrent.cpp

```diff
@@ -81,48 +101,31 @@ void test_running_torrent(boost::shared_ptr<torrent_info> info, boost::int64_t f
 	}
 
 	std::vector<int> ones(info->num_files(), 1);
-	h.prioritize_files(ones);
+	TEST_CHECK(prioritize_files(h, ones))
 
-//	test_sleep(500);
+	// test_sleep(500);
 	torrent_status st = h.status();
 
 	TEST_EQUAL(st.total_wanted, file_size); // we want the single file
 	TEST_EQUAL(st.total_wanted_done, 0);
 
 	std::vector<int> prio(info->num_files(), 1);
 	prio[0] = 0;
-	h.prioritize_files(prio);
-	st = h.status();
+	TEST_CHECK(prioritize_files(h, prio))
 
+	st = h.status();
 	TEST_EQUAL(st.total_wanted, 0); // we don't want anything
 	TEST_EQUAL(st.total_wanted_done, 0);
 	TEST_EQUAL(int(h.file_priorities().size()), info->num_files());
-	if (!st.is_seeding)
-	{
-		TEST_EQUAL(h.file_priorities()[0], 0);
-		if (info->num_files() > 1)
-			TEST_EQUAL(h.file_priorities()[1], 1);
-		if (info->num_files() > 2)
-			TEST_EQUAL(h.file_priorities()[2], 1);
-	}
 
 	if (info->num_files() > 1)
 	{
 		prio[1] = 0;
-		h.prioritize_files(prio);
-		st = h.status();
+		TEST_CHECK(prioritize_files(h, prio))
 
+		st = h.status();
 		TEST_EQUAL(st.total_wanted, file_size);
 		TEST_EQUAL(st.total_wanted_done, 0);
-		if (!st.is_seeding)
-		{
-			TEST_EQUAL(int(h.file_priorities().size()), info->num_files());
-			TEST_EQUAL(h.file_priorities()[0], 0);
-			if (info->num_files() > 1)
-				TEST_EQUAL(h.file_priorities()[1], 0);
-			if (info->num_files() > 2)
-				TEST_EQUAL(h.file_priorities()[2], 1);
-		}
```

Comment:
I think these tests should still be here, or is there a reason to remove them?

Reply:
Everything that is being tested here is now covered by `TEST_CHECK(prioritize_files(h, prio))`, plus it also checks the rest of priorities that should have been set.